config/config.go

```diff
@@ -62,26 +81,147 @@ func TryLoadRootConfig(fromdir string) (cfg hcl.Config, configpath string, found
 		}
 		fromdir = parent
 	}
-	return hcl.Config{}, "", false, nil
+	return nil, "", false, nil
 }
 
-// IsStack returns true if the given directory is a stack, false otherwise.
-func IsStack(rootdir, dir string) (bool, error) {
+func LoadTree(rootdir string, cfgdir string) (*Tree, error) {
+	return loadTree(rootdir, cfgdir, nil)
+}
+
+func (tree *Tree) Rootdir() string {
+	return tree.rootdir
+}
+
+func (tree *Tree) IsStack() bool {
+	return tree.Root.Stack != nil
+}
+
+func (tree *Tree) Stacks() []*Tree {
```

Comment:
```suggestion
Stacks returns all Tree nodes that are stacks.
func (tree *Tree) Stacks() []*Tree {
```

Reply:
thanks, I left docs to the final. I still need to go over all of them and adjust.

Reply:
I imagined something like that, so I tried to help a little with them.